jina/clients/base/helper.py

```diff
@@ -147,7 +147,7 @@ async def send_message(self, request: 'Request'):
                 r_str = await response.json()
                 handle_response_status(response.status, r_str, self.url)
                 return response
-            except (ValueError, ConnectionError, BadClient, aiohttp.ClientError) as err:
+            except (ValueError, ConnectionError, BadClient, aiohttp.ClientError, aiohttp.ContentTypeError) as err:
```

Comment:
Why don't we check the content-type to extract the `r_str`. Here the status code is more important or should be checked first before extracting the payload. The underlying issue is a `504` status code with some default response from the Ingress component which is used raise a new error. You can check the `handle_response_status` method also which already handles the `504` http error code.

Reply:
You are right, I am going to check the status code prior to decoding.